pkg/ingester/ingester.go

```diff
@@ -138,6 +141,18 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.SpreadFlushes, "ingester.spread-flushes", false, "If true, spread series flushes across the whole period of MaxChunkAge")
 	f.IntVar(&cfg.ConcurrentFlushes, "ingester.concurrent-flushes", 50, "Number of concurrent goroutines flushing to dynamodb.")
 	f.DurationVar(&cfg.RateUpdatePeriod, "ingester.rate-update-period", 15*time.Second, "Period with which to update the per-user ingestion rates.")
+
+	// Prometheus 2.x block storage
+	f.BoolVar(&cfg.V2.Enabled, "ingester.v2.enable", false, "If true, use Prometheus block storage for metrics")
+	f.DurationVar(&cfg.V2.BlockRanges, "ingester.v2.blockranges", 1*time.Hour, "TSDB block ranges")
+	f.DurationVar(&cfg.V2.Retention, "ingester.v2.retention", 6*time.Hour, "TSDB block retention")
+	f.StringVar(&cfg.V2.TSDBDir, "ingester.v2.dir", "tsdb", "directory to place all tsdb's into")
+	f.StringVar(&cfg.V2.S3Key, "ingester.v2.key", "", "s3 skey")
```

Comment:
Without adding too much extra complexity to this PR, I would suggest to move S3 settings under a dedicated `aws-s3` config block, in order to ease adding more backend storages in the future, just because refactoring config file / CLI flags later on is way more complicated (ie. after this PR will get merged, I would like to offer a PR to add GCS support).

An idea is:
```
tsdb:
    backend: s3
    s3:
        access_key_id:
        secret_access_key:
        bucket_name:
        endpoint:
```

CLI flags should be specular, with `_`  replaced by `-` (ie. `tsdb.s3.access-key-id`). What's your take?

Reply:
resolved with your merged changes.